lib/index.js

```diff
@@ -22,6 +22,7 @@ module.exports = function (dir) {
     whitelist: ['NODE_ENV', 'PORT'],
   });
   nconf.argv();
+  nconf.file('secret', { file: path.resolve(secretConfig) });
```

Comment:
Placing this here sets the right priority for the `secret` config.
I am using the `no-such-file` default value because using an empty value fails (causes reading from a dir). It is too complex to have this layer added in an `if` statement such as
```
if (secretConfig) {
  nconf.file('secret', { file: path.resolve(secretConfig) });
}
```
because `nconf` is global, and the order of layers is set with the first test, and is not reset between tests.